pytorch_lightning/distributed/dist.py

```diff
@@ -18,6 +18,12 @@
 
 
 class LightningDistributed:
+    """
+    .. deprecated:: v1.5
+        This method is deprecated in v1.5and will be removed in v1.7.
+        Logic moved to ddp and ddpspawning
+    """
+
     def __init__(self, rank=None, device=None):
```

Comment:
1. add a deprecation message in the init so a warning shows up at runtime
2. Add a test for the deprecation here: https://github.com/PyTorchLightning/pytorch-lightning/blob/master/tests/deprecated_api/test_remove_1-7.py

so we know to delete this class in v1.7

Reply:
@awaelchli @ananthsub  This is a internal use only class, do you guys think we need wait until 1.7 to remove this or it is safe to be directly removed?

Reply:
yes I think we need to unfortunately. it's been around for a while and is not marked as experimental.

Reply:
we can remove its usage from within the plugins, but we cannot delete the class outright. it will have to sit unused in the codebase until v1.7

Reply:
@ananthsub's comment reminds me that we should then exclude this file also from coverage. we can add it in `setup.cfg ` and add it to the list of omitted files